.github/workflows/main.yml

```diff
@@ -36,21 +42,92 @@ jobs:
+        uses: docker/build-push-action@v5 # https://github.com/marketplace/actions/build-and-push-docker-images
+        with:
+          context: workspaces/desktop
+          platforms: linux/amd64,linux/arm64
+          push: ${{ env.CAN_PUSH_IMAGE == 'true' }}
+          no-cache: true
+          tags: prairielearn/workspace-desktop:${{ env.COMMIT_SHA }}
+
+      ######################################################################################################
+      # prairielearn/workspace-jupyterlab
+      - name: Check whether workspaces/jupyterlab was modified
+        run: tools/check_path_modified.sh workspaces/jupyterlab workspaces_jupyterlab_modified
+      - name: Build and push prairielearn/workspace-jupyterlab
+        if: ${{ env.workspaces_jupyterlab_modified }}
+        uses: docker/build-push-action@v5 # https://github.com/marketplace/actions/build-and-push-docker-images
+        with:
+          context: workspaces/jupyterlab
+          platforms: linux/amd64,linux/arm64
+          push: ${{ env.CAN_PUSH_IMAGE == 'true' }}
+          no-cache: true
+          tags: prairielearn/workspace-jupyterlab:${{ env.COMMIT_SHA }}
+
+      ######################################################################################################
+      # prairielearn/workspace-jupyterlab-python
+      - name: Check whether workspaces/jupyterlab-python was modified
+        run: tools/check_path_modified.sh workspaces/jupyterlab-python workspaces_jupyterlab_python_modified
+      - name: Build and push prairielearn/workspace-jupyterlab-python
+        if: ${{ env.workspaces_jupyterlab_python_modified }}
+        uses: docker/build-push-action@v5 # https://github.com/marketplace/actions/build-and-push-docker-images
+        with:
+          context: workspaces/jupyterlab-python
+          platforms: linux/amd64,linux/arm64
+          push: ${{ env.CAN_PUSH_IMAGE == 'true' }}
+          no-cache: true
+          tags: prairielearn/workspace-jupyterlab-python:${{ env.COMMIT_SHA }}
+
+      ######################################################################################################
+      # prairielearn/workspace-rstudio
+      - name: Check whether workspaces/rstudio was modified
+        run: tools/check_path_modified.sh workspaces/rstudio workspaces_rstudio_modified
+      - name: Build and push prairielearn/workspace-rstudio
+        if: ${{ env.workspaces_rstudio_modified }}
+        uses: docker/build-push-action@v5 # https://github.com/marketplace/actions/build-and-push-docker-images
+        with:
+          context: workspaces/rstudio
+          platforms: linux/amd64,linux/arm64
+          push: ${{ env.CAN_PUSH_IMAGE == 'true' }}
+          no-cache: true
+          tags: prairielearn/workspace-rstudio:${{ env.COMMIT_SHA }}
 
-      - name: Build the workspace-desktop docker image if needed
-        run: ./tools/build-image-if-needed.sh workspaces/desktop prairielearn/workspace-desktop:latest
```

Comment:
Can we delete the `build-image-if-needed.sh` script?

Reply:
There was one more usage of this relating to `prairielearn/plbase`. I changed that build to use the Docker Action in fcbbc9f, which left me able to delete that script 🎉 

I committed an empty line to `images/plbase/Dockerfile` to force a rebuild to test the change. I'll back that out before I merge.